templates/tfengine/components/project/project/main.tf

```diff
@@ -1,28 +1,47 @@
-# Copyright 2020 Google Inc.
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
+{{- /* Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */ -}}
 
 # Create the project, enable APIs, and create the deletion lien, if specified.
 module "project" {
   source  = "terraform-google-modules/project-factory/google"
-  version = "~> 7.0"
+  version = "~> 8.0.0"
 
-  name                    = var.name
+  name                    = "{{.PROJECT_ID}}"
   org_id                  = var.org_id
   folder_id               = var.folder_id
   billing_account         = var.billing_account
-  lien                    = var.enable_lien
-  activate_apis           = var.apis
+  lien                    = {{get . "ENABLE_LIEN" "true"}}
   default_service_account = "keep"
   skip_gcloud_download    = true
+
+  {{- if has . "SHARED_VPC"}}
+  shared_vpc              = "{{.SHARED_VPC.HOST}}"
+
+  {{- if has . "SHARED_VPC.SUBNETS"}}
+  shared_vpc_subnets = [
+    {{- range get . "SHARED_VPC.SUBNETS"}}
+    "{{.}}",
+    {{- end}}
+  ]
+  {{- end}}
+  {{- end}}
```

Comment:
Add comments on which block these close out.

Reply:
Do you have an example of what to put here? I have only seen this in C++ type languages and I personally feel if we strictly follow indentation then it's okay to visually find the brace. But if you feel it helps readabilty I can add comments.

Reply:
I think it's a good idea here only because these are both indented at the same level. I think just something like `{{- /* end SHARED_VPC.SUBNETS */-}}` and `{{- /* end SHARED_VPC */-}}` is fine.

Reply:
Normally I agree, I also don't like "end X" comments for things that are otherwise visually indented.

Reply:
Good point, I think it definitely helps here. Added comments